### PR TITLE
More proper approach to get response body

### PR DESCRIPTION
**Pull request by L11R**
We had unnecessary memory transfusions

### CODE REVIEW

File: ovrstat/player_stats.go [hunk 1]
```diff
@@ -78,18 +76,16 @@ func playerStats(profilePath string, platform string) (*PlayerStats, error) {
...
 	}
-	defer res.Body.Close()
-	body, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		return nil, err
-	}
 
 	// Parses the stats request into a goquery document
-	pd, err := goquery.NewDocumentFromReader(bytes.NewReader(body))
+	pd, err := goquery.NewDocumentFromReader(res.Body)
```
Comment (s32x) [inline ovrstat/player_stats.go]: I like this, but if it fails to create a goquery document the reader will never be closed. We need to defer close before this.

File: ovrstat/player_stats.go [hunk 2]
```diff
@@ -120,11 +121,15 @@ func playerStats(profilePath string, platform string) (*PlayerStats, error) {
...
 	}
-	defer res.Body.Close()
+
+	// Decode received JSON
 	if err := json.NewDecoder(res.Body).Decode(&platforms); err != nil {
 		return nil, ErrPlayerNotFound
 	}
 
+	// Close it again
+	res.Body.Close()
```
Comment (s32x) [inline ovrstat/player_stats.go]: Same here, the body's need to be defer closed before they are used. If you're afraid of rewriting `res` we could perhaps rename the response variable to something other than `res` and defer close that instead.

Comment (L11R) [reply ovrstat/player_stats.go]: @s32x yeah, sounds like a solution. Will send fix a little bit later.

**Comment by s32x** [comment]
This is really good, For some reason, I just always decode the body to bytes before unmarshaling. I don't know why but it has never occurred to me to use the reader. Thanks for this!

**Comment by L11R** [comment]
@s32x yeah I started to use readers only after short dive into low level world :D Cryptography, image, sound and video decoding, readers/writers and io API are everywhere.

**Pull request merged**